src/role_carry.js

```diff
@@ -22,6 +22,18 @@ roles.carry.settings = {
 };
 
 roles.carry.preMove = function(creep, directions) {
+  // Fix blocked helpers due to empty structure in the room where we get the energy from
+  if (creep.room.name === creep.memory.routing.targetRoom) {
+    let targetStructure = Game.getObjectById(creep.memory.routing.targetId);
+    if (targetStructure.structureType == STRUCTURE_STORAGE) {
+      creep.say('storage');
+      if (targetStructure.store.energy === 0) {
+        creep.log('Suiciding the storage I should get the energy from is empty');
+        creep.suicide();
+      }
```

Comment:
it should be better to place it into canHelp definitiotn i supose. i'll look at.